src/components/fields/image-upload/image-input/file-item/file-item.tsx

```diff
@@ -0,0 +1,171 @@
+import {
+	CellDeleteButton,
+	CellFileSize,
+	CellInfo,
+	CellProgressBar,
+	DeleteButton,
+	ErrorText,
+	MobileFileSize,
+	ProgressBar,
+	TextBody,
+	Thumbnail,
+	Wrapper,
+} from "./file-item.styles";
+
+interface IProps {
+	id?: string;
+	index: number;
+	fileItem: IImage;
+	maxSize: number;
+	accepts: TImageUploadAcceptedFileType[];
+	validation: IImageUploadValidationRule[];
+	onDelete: (index: number) => (event: React.MouseEvent<HTMLButtonElement>) => void;
+}
+
+export const FileItem = ({ id = "file-item", index, fileItem, maxSize, accepts, onDelete, validation }: IProps) => {
+	// =============================================================================
+	// CONST, STATE, REFS
+	// =============================================================================
+	const { dataURL, file, name: fileName, status, uploadProgress } = fileItem;
+	const [isError, setError] = useState<boolean>(false);
+	const [errorMessage, setErrorMessage] = useState<string>();
+	const fileNameWrapperRef = useRef<HTMLDivElement>(null);
+	const [transformedFileName, setTransformedFileName] = useState<string>();
+	// =============================================================================
+	// HELPER FUNCTIONS
+	// =============================================================================
+	const setFileNameToWidth = useCallback(() => {
+		let widthOfElement = 0;
+		if (fileNameWrapperRef && fileNameWrapperRef.current) {
+			widthOfElement = fileNameWrapperRef.current.getBoundingClientRect().width;
+		}
+		const transformed = FileHelper.truncateFileName(fileName, widthOfElement);
+		setTransformedFileName(transformed);
+	}, [fileName]);
+
+	useEffect(() => {
+		window.addEventListener("resize", setFileNameToWidth);
+		return () => window.removeEventListener("resize", setFileNameToWidth);
+	}, [setFileNameToWidth]);
+
+	// =============================================================================
+	// EFFECTS
+	// =============================================================================
+	useEffect(() => {
+		switch (status) {
+			case EImageStatus.ERROR_FORMAT: {
+				const fileTypeRule = validation?.find((rule) => "fileType" in rule);
+				const errorMessage = fileTypeRule?.errorMessage || ERROR_MESSAGES.UPLOAD("photo").FILE_TYPE(accepts);
+				setError(true);
+				setErrorMessage(errorMessage);
+				break;
+			}
+			case EImageStatus.ERROR_GENERIC:
+				setError(true);
+				setErrorMessage(ERROR_MESSAGES.UPLOAD("photo").GENERIC);
+				break;
+			case EImageStatus.ERROR_SIZE: {
+				const fileSizeRule = validation?.find((rule) => "maxSize" in rule);
+				const errorMessage =
+					fileSizeRule?.errorMessage || ERROR_MESSAGES.UPLOAD("photo").MAX_FILE_SIZE(maxSize);
+				setError(true);
+				setErrorMessage(errorMessage);
+				break;
+			}
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [status, dataURL, file.type, maxSize]);
```

Comment:
why `dataUrl` and `file.type`?

Reply:
have removed it, it could have been part of a removed logic but I don't see it is needed now